quic/s2n-quic-platform/src/message/msg.rs

```diff
@@ -253,7 +258,7 @@ pub(super) unsafe fn alloc<T: Copy + Sized, F: Fn(&mut T) -> &mut msghdr>(
             let entry = on_entry((*entry_ptr).get_mut());
             (*header_ptr)
                 .get_mut()
-                .update(entry, &*payload_ptr, payload_len);
+                .update(entry, (*payload_ptr).get_mut(), payload_len);
```

Comment:
I don't think this is what you want -- update wants a *mut u8 which is "valid" for the full range of the payload. I think `.cast::<u8>()` is better here, or keep payload_ptr as `*mut u8` -- the UnsafeCell should I think never be needed, since I don't recall us going from &T to &mut T anywhere or similar?